src/packagedcode/maven.py

```diff
@@ -145,6 +146,9 @@ def api_data_url(self, baseurl=default_api_baseurl):
             filename=filename,
             baseurl=baseurl)
 
+    def compute_normalized_license(self):
+        declared_license_str = json.dumps(self.declared_license)
```

Comment:
I am not sure we want to use a straight dump. Instead the whole idea is to detect on the filed values and not the whole thing at once which is the problem we have right now. Did you add some tests of problematic POMs?

Reply:
Good idea.  We didn't carry the original data in this code, I would use declared_license to carry the original data and then it's easy to pass the normalized_license as the original data. That would be better for the license detection.

This is the our discussion,  2 or 3 is the best solution.

1. keep it as an unstructured multiline strings as we have today
2. store it as JSON carrying the structure that was in the original manifest
3. store it as a string exactly as it was in the original manifest: an XML snippet for a POM, a JSON snippet for an npm, etc

Reply:
Thanks for the comment, I got the idea and have some ideas, I will update and let  you review again.

Reply:
> Good idea. We didn't carry the original data in this code, I would use declared_license to carry the original data and then it's easy to pass the normalized_license as the original data. That would be better for the license detection.
> 
> This is the our discussion, 2 or 3 is the best solution.
> 
>     1. keep it as an unstructured multiline strings as we have today
> 
>     2. store it as JSON carrying the structure that was in the original manifest
> 
>     3. store it as a string exactly as it was in the original manifest: an XML snippet for a POM, a JSON snippet for an npm, etc

we need 2. alright ... but the whole point is that we want to detect these in sequence for the maven case:
1. the name
2. the url. If url is not properly detected ignore url. If fully detected but with a different license than name, combine the results of the two in an expression
3. the comment, with a process similar to the url.

We need to do this for all the <licenses>

Later for npms, the idea would be the same using `license` and `licenses.type` and `licenses.url` instead